pkg/devfile/convert/convert.go

```diff
@@ -236,6 +242,52 @@ func setDevfileCommandsForS2I(d data.DevfileData) {
 
 }
 
+func setInitContainer(d data.DevfileData, s2iImage string) error {
+	err := d.AddEvents(devfilev1.Events{
+		DevWorkspaceEvents: devfilev1.DevWorkspaceEvents{
+			PreStart: []string{"copy-app-root"},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	err = d.AddCommands([]devfilev1.Command{
+		{
+			Id: "copy-app-root",
+			CommandUnion: devfilev1.CommandUnion{
+				Apply: &devfilev1.ApplyCommand{
+					Component: "copy-app-root-container",
+				},
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	initContainer := devfilev1.Component{
+		Name: "copy-app-root-container",
+		ComponentUnion: devfilev1.ComponentUnion{
+			Container: &devfilev1.ContainerComponent{
+				Container: devfilev1.Container{
+					Image: s2iImage,
+					VolumeMounts: []devfilev1.VolumeMount{
+						{
+							Name: "app-root-volume",
+							Path: "/mnt/app-root",
+						},
+					},
+					Command: []string{
+						"/bin/sh",
+						"-c",
+						"[ -d /opt/app-root ] && cp -R /opt/app-root /mnt/ || true",
```

Comment:
Yes, the script will always return true, even if some error occurs during copying the files. This is to prevent the initContainer to finish as Error, that would terminate the pod immediately